core/src/main/java/com/linecorp/armeria/common/RequestContext.java

```diff
@@ -625,4 +629,24 @@ default Runnable makeContextAware(Runnable runnable) {
     default Logger makeContextAware(Logger logger) {
         return ContextAwareLogger.of(this, logger);
     }
+
+    /**
+     * Returns an {@link ExecutorService} that will execute callbacks in the given {@code executor}, propagating
+     * the caller's {@link RequestContext} (if any) into the callback execution.
+     * If this executor service is only used from a single request then it's better to use
+     * {@link #makeContextAware(ExecutorService)}
+     */
+    static ExecutorService makeContextPropagating(ExecutorService executor) {
```

Comment:
Last but not least, please group static methods with other static methods at the top. How about right next to `builder()`?